.travis.yml

```diff
@@ -18,149 +18,171 @@ branches:
     - /^\d+\.\d+\.x$/
     - /^test-.*$/
 
+# Normal job always executed (including PR) except pushes on master
+not-on-master: &not-on-master
+  if: NOT (type = push AND branch = master)
+
+# Extended job executed on cron jobs and pushes on non-master branches
+extended-pipeline: &extended-pipeline
+  if: type = cron OR (type = push AND branch != master)
+
 matrix:
   include:
-    # These environments are always executed
+    # Main pipeline
     - python: "2.7"
       env: BOULDER_INTEGRATION=v1 INTEGRATION_TEST=all TOXENV=py27_install
       sudo: required
       services: docker
+      <<: *not-on-master
     - python: "2.7"
       env: BOULDER_INTEGRATION=v2 INTEGRATION_TEST=all TOXENV=py27_install
       sudo: required
       services: docker
+      <<: *not-on-master
+
+    # This job is always executed, including on master
     - python: "2.7"
       env: TOXENV=py27-cover FYI="py27 tests + code coverage"
+
     - sudo: required
       env: TOXENV=nginx_compat
       services: docker
       before_install:
       addons:
+      <<: *not-on-master
     - python: "2.7"
       env: TOXENV=lint
+      <<: *not-on-master
     - python: "3.4"
       env: TOXENV=mypy
+      <<: *not-on-master
     - python: "3.5"
       env: TOXENV=mypy
+      <<: *not-on-master
     - python: "2.7"
       env: TOXENV='py27-{acme,apache,certbot,dns,nginx,postfix}-oldest'
       sudo: required
       services: docker
+      <<: *not-on-master
     - python: "3.4"
       env: TOXENV=py34
       sudo: required
       services: docker
+      <<: *not-on-master
     - python: "3.7"
       dist: xenial
       env: TOXENV=py37
       sudo: required
       services: docker
+      <<: *not-on-master
     - sudo: required
       env: TOXENV=apache_compat
       services: docker
       before_install:
       addons:
+      <<: *not-on-master
     - sudo: required
       env: TOXENV=le_auto_trusty
       services: docker
       before_install:
       addons:
+      <<: *not-on-master
     - python: "2.7"
       env: TOXENV=apacheconftest-with-pebble
       sudo: required
       services: docker
+      <<: *not-on-master
     - python: "2.7"
       env: TOXENV=nginxroundtrip
+      <<: *not-on-master
 
-    # These environments are executed on cron events and commits to tested
-    # branches other than master. Which branches are tested is controlled by
-    # the "branches" section earlier in this file.
+    # Extended pipeline on cron jobs and push to test-* branches
```

Comment:
nit: I'd change this to something like "and pushes to tested branches other than master". Another important category here are point release branches.

Reply:
Done.